pkg/agent/agent.go

```diff
@@ -338,7 +349,20 @@ func (c *agentController) generateExtManagedKubeconfigSecret(ctx context.Context
 	klusterletNamespace := &corev1.Namespace{}
 	klusterletNamespaceNsn := types.NamespacedName{Name: "klusterlet-" + managedClusterAnnoValue}
 
-	err := c.spokeClient.Get(ctx, klusterletNamespaceNsn, klusterletNamespace)
+	//When klusterlet for imported HC is created, namespace might not exist right away, attempt twice before failing
+	var err error
+	for try := 1; try <= 5; try++ {
+		if try != 1 {
+			c.log.Error(err, 
```

Comment:
Maybe make this an Info message instead of an error. After the 5 attempts failed, then we log an error msg